mne/io/base.py

```diff
@@ -687,17 +687,15 @@ def set_annotations(self, annotations, emit_warning=True,
                                    ' which case the annotation onsets would be'
                                    ' taken in reference to the first sample of'
                                    ' the raw object.')
-
-            delta = 1. / self.info['sfreq']
```

Comment:
According to the CI test results, it seems that removing this delta will affect the annotation filtering behavior. For example:
```
data = np.ones((1, 1000))
info = create_info(1, 1000., 'eeg')
raw = RawArray(data, info)
```

this will result in data duration of 0.999 instead of 1 second.

Reply:
So it seems like maybe we do need to keep `delta` in there for deciding which annotation `duration`s to crop, but adjust which annotations get removed based on `onset` ...? Like in the case of 1 sec of data at 1000 Hz, if an annotation has `onset=0` and duration = `1` it shouldn't be cropped, but one with an `onset >= 1` should be removed because it actually starts (rather than ends) at the next time point?